src/common/snippets/src/lowered/pass/init_live_ranges.cpp

```diff
@@ -0,0 +1,87 @@
+#include "snippets/lowered/expressions/buffer_expression.hpp"
+
+namespace ov {
+namespace snippets {
+namespace lowered {
+namespace pass {
+namespace {
+// Expressions that don't affect lifetime of registers, e.g. Buffer or RankNormalization
+inline bool pass_through_expr(const ExpressionPtr& expr) {
+    const auto& node = expr->get_node();
+    return op::Subgraph::is_shape_infer_op(node)
+#ifdef SNIPPETS_DEBUG_CAPS
+            || ov::is_type<op::PerfCountBeginBase>(node)
+            || ov::is_type<op::PerfCountEndBase>(node)
+#endif
+            || ov::is_type<BufferExpression>(expr);
+}
+
+} // namespace
+
+bool InitLiveRanges::run(LinearIR& linear_ir) {
+    OV_ITT_SCOPED_TASK(ov::pass::itt::domains::SnippetsTransform, "Snippets::InitLiveRanges")
+    std::map<RegType, size_t> reg_counter;
+
+    // Note: map expiring time to register
+    std::map<double, std::set<Reg>> regs_to_expire;
+
+    for (auto expr_it = linear_ir.begin(); expr_it != linear_ir.end(); expr_it++) {
+        const auto& expr = *expr_it;
+        const auto op = expr->get_node();
+        if (pass_through_expr(expr)) {
+            expr->set_live_regs(std::prev(expr_it)->get()->get_live_regs());
+            continue;
+        }
+        const double start = expr->get_exec_num();
+        // Remove all regs that expired before start
+        regs_to_expire.erase(regs_to_expire.begin(), regs_to_expire.lower_bound(start)); // remove all elements lower than start (not equal)
+        std::set<Reg> live_regs;
+        for (const auto& time_reg : regs_to_expire)
+            live_regs.insert(time_reg.second.begin(), time_reg.second.end());
+
+        expr->set_live_regs(std::move(live_regs));
+
+        for (size_t i = 0; i < expr->get_output_count(); ++i) {
+            const auto& out_pd = expr->get_output_port_descriptor(i);
+            if (out_pd->get_reg().is_defined())
+                continue;
+            const auto reg_type = m_reg_manager.get_reg_type(op->output(i));
+            const auto& reg = Reg(reg_type, reg_counter[reg_type]++);
+            double stop = start;
+            // propagate to consumers
+            std::stack<PortConnectorPtr> to_visit;
+            to_visit.push(expr->get_output_port_connector(i));
+            while (!to_visit.empty()) {
+                const auto& current = to_visit.top();
+                current->get_source().get_descriptor_ptr()->set_reg(reg);
+                to_visit.pop();
+                for (const auto& consumer : current->get_consumers()) {
+                    consumer.get_descriptor_ptr()->set_reg(reg);
+                    const auto& consumer_expr = consumer.get_expr();
+                    stop = std::max(stop, consumer_expr->get_exec_num());
+                    if (pass_through_expr(consumer_expr)) {
+                        for (const auto& connector : consumer_expr->get_output_port_connectors())
+                            to_visit.push(connector);
+                    }
```

Comment:
May I ask you to add a comment on why we need to propagate reg info through the `pass_through_expr`?

Reply:
No problem, added